CARDS-2023 - PREMs: Rehab survey configuration

marta- opened the pull request.

marta- committed:
CARDS-2023 - PREMs: Rehab survey configuration

marta- committed:
CARDS-2023 - PREMs: Rehab survey configuration
Added side bar entry for Rehab Dashboard

sashaandjic commented:
The new Rehab survey has been added as an option on the visit form. Should we also add Rehab option on the dashboard?
![image](https://user-images.githubusercontent.com/23243244/214792391-5dc4d798-a631-4e0b-89c3-012f75399359.png)
![image](https://user-images.githubusercontent.com/23243244/214792326-c19b32de-0419-4770-a603-a873c032fe1b.png)
![image](https://user-images.githubusercontent.com/23243244/214792352-cfd27443-86bc-4d8f-ac8e-db7eb9a2b1b3.png)

marta- commented:
> The new Rehab survey has been added as an option on the visit form. Should we also add Rehab option on the dashboard?

Right, I forgot about the dashboard because nobody will be using it anyway. I'll add the entry for consistency.

This PR also includes the email templates for Rehab, and someone needs to check and test those as well.

sashaandjic commented:
WSOD when clicking on the newly added Rehab
![image](https://user-images.githubusercontent.com/23243244/214894505-3a0a54e4-b5e8-4838-9f87-240d3a9cdf21.png)

marta- commented:
> WSOD when clicking on the newly added Rehab

Sorry about that, there was a typo in the QuestionnaireSet causing the columns definition for the Rehab survey box to have malformed syntax. Fixed by amending the first commit for a cleaner history.

sashaandjic commented:
tested all but email templates
![image](https://user-images.githubusercontent.com/23243244/214934475-eebe3faf-45ef-4319-ad76-b1ffb4009e38.png)

acrowthe commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml:
```diff
@@ -0,0 +1,2290 @@
+				<value>hidden</value>
+				<type>String</type>
+			</property>
+		</node>
```
entryMode reference will need to be added here once 1887 is merged

marta- replied:
Thanks for catching this, I thought it was already there just like for department. Adding now.

marta- replied:
Done in 95c376e.

marta- committed:
CARDS-2023 - PREMs: Rehab survey configuration

Added missing entryMode: reference + reference question to the hidden hospital field

sashaandjic reviewed (approved):
tested - Provider and Location do get copied over to Rehab survey from Visit info form

sashaandjic commented:
![image](https://user-images.githubusercontent.com/23243244/214951136-84b952b0-482d-44c5-83a0-9853d484a708.png)
![screencapture-localhost-8080-Survey-html-2023-01-26-16_10_05](https://user-images.githubusercontent.com/23243244/214951606-92605a14-2569-4df3-a063-a9e11f5b7048.png)

sashaandjic reviewed (approved):
tested the following:
- new rehab surveys form
- new rehab dashboard button
- Provider and Location copied over from Visit info form to Rehab survey form (manually created Rehab form)
- checked the email template
- checked that email link is working
- filled out rehab survey from provided link
- submitted rehab survey form

acrowthe commented:
Tested reference questions rebased onto 1887 - also working with autocreated forms
